ui/workspaceMonitor.js

```diff
@@ -15,8 +15,9 @@ class WorkspaceMonitor extends GObject.Object {
         this._shellwm.connect('minimize', this._windowDisappearing.bind(this));
         this._shellwm.connect('destroy', this._windowDisappearing.bind(this));
 
-        global.window_group.connect('actor-added', this._windowsChanged.bind(this));
-        global.window_group.connect('actor-removed', this._windowsChanged.bind(this));
+        this._windowGroup = global.window_group;
+        this._windowGroup.connect('actor-added', this._windowsChanged.bind(this));
+        this._windowGroup.connect('actor-removed', this._windowsChanged.bind(this));
```

Comment:
I think we should add a destroy() method/handler here and disconnect all these global signals although I see this (WorkspaceMonitor) object is not destroyed when we disable the extension I think it is good practice to make sure the lifecycle of the objects are properly handled. (I see we already check for `this._enabled` in the relevant callbacks but that wouldn't work if we start destroying this object on `disable()` which I dont see why not?).

Reply:
I mean, not only the window_group signals here but also the other global signals we handle.